src/msft-recommendation/src/test/scala/TrainValidRecommendSplitRec.scala

```diff
@@ -0,0 +1,596 @@
+import org.apache.spark.ml.tuning.ParamGridBuilder
+import org.apache.spark.ml.util.MLReadable
+import org.apache.spark.sql.DataFrame
+
+import scala.language.existentials
+import scala.tools.nsc.interpreter.session
+
+class TrainValidRecommendSplitRec
+  extends TestBase with EstimatorFuzzing[TrainValidRecommendSplit] {
+
+  override def testObjects(): Seq[TestObject[TrainValidRecommendSplit]] = {
+    val df: DataFrame = session
+      .createDataFrame(Seq(
+        ("11", "Movie 01", 4),
+        ("11", "Movie 03", 1),
+        ("11", "Movie 04", 5),
+        ("11", "Movie 05", 3),
+        ("11", "Movie 06", 4),
+        ("11", "Movie 07", 1),
+        ("11", "Movie 08", 5),
+        ("11", "Movie 09", 3),
+        ("22", "Movie 01", 4),
+        ("22", "Movie 02", 5),
+        ("22", "Movie 03", 1),
+        ("22", "Movie 05", 3),
+        ("22", "Movie 06", 4),
+        ("22", "Movie 07", 5),
+        ("22", "Movie 08", 1),
+        ("22", "Movie 10", 3),
+        ("33", "Movie 01", 4),
+        ("33", "Movie 03", 1),
+        ("33", "Movie 04", 5),
+        ("33", "Movie 05", 3),
+        ("33", "Movie 06", 4),
+        ("33", "Movie 08", 1),
+        ("33", "Movie 09", 5),
+        ("33", "Movie 10", 3),
+        ("44", "Movie 01", 4),
+        ("44", "Movie 02", 5),
+        ("44", "Movie 03", 1),
+        ("44", "Movie 05", 3),
+        ("44", "Movie 06", 4),
+        ("44", "Movie 07", 5),
+        ("44", "Movie 08", 1),
+        ("44", "Movie 10", 3)))
+      .toDF("customerIDOrg", "itemIDOrg", "rating")
+
+    val customerIndex: StringIndexer = new StringIndexer()
+      .setInputCol("customerIDOrg")
+      .setOutputCol("customerID")
+
+    val ratingsIndex: StringIndexer = new StringIndexer()
+      .setInputCol("itemIDOrg")
+      .setOutputCol("itemID")
+
+    val pipeline = new Pipeline()
+      .setStages(Array(customerIndex, ratingsIndex))
+
+    val transformedDf = pipeline.fit(df).transform(df)
+
+    val alsWReg = new MsftRecommendation()
+      .setUserCol(customerIndex.getOutputCol)
+      .setRatingCol("rating")
+      .setItemCol(ratingsIndex.getOutputCol)
+
+    val paramGrid = new ParamGridBuilder()
+      .addGrid(alsWReg.regParam, Array(1.0))
+      .build()
+
+    val evaluator = new MsftRecommendationEvaluator()
+      .setK(3)
+    val tvRecommendationSplit = new TrainValidRecommendSplit()
+      .setEstimator(alsWReg)
+      .setEvaluator(evaluator)
+      .setEstimatorParamMaps(paramGrid)
+      .setTrainRatio(0.8)
+      .setUserCol(customerIndex.getOutputCol)
+      .setRatingCol("rating")
+      .setItemCol(ratingsIndex.getOutputCol)
+    //    val testDF = pipeline.fit(df).transform(df)
+
+    List(new TestObject(tvRecommendationSplit, transformedDf))
+  }
+
+  override def reader: MLReadable[_] = TrainValidRecommendSplit
+
+  override def modelReader: MLReadable[_] = TrainValidRecommendSplitModel
+
+//  test("testALSPipeline") {
+//    val dfRaw2: DataFrame = session
```

Comment:
Maybe remove this? 
Alternatively for large dataset, you can move it to a text file and parse it inside the testcase.